### DISCUSSION TITLE
[Support] 升级为v3去除对 winston 的依赖后，如何把 OpenTelemetry 的 traceId 打印到日志中

### DISCUSSION THREAD

**Discussion by Yuliang-Lee** [midwayjs/midway#3725] [Q&A]
### Describe the problem（描述问题）

原来引入 `@opentelemetry/instrumentation-winston`的话可以在日志输出中加入 traceId，现在改为自研实现的 logger 如何在日志里插入 openTelemetry 的 traceId？

### Midway Versions（Midway 版本）

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(not installed)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.14.0)
✓ @midwayjs/bull(3.14.0)
✓ @midwayjs/bull-board(3.14.0)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/cache-manager-redis(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.14.0)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(3.14.0)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(3.14.0)
✓ @midwayjs/info(not installed)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.14.0)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✖ @midwayjs/redis(current: 3.13.9, allow: ["3.14.0"])
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(not installed)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(3.14.0)
✓ @midwayjs/upload(not installed)
✓ @midwayjs/validate(3.14.0)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(3.14.0)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(not installed)
✓ @midwayjs/ws(not installed)

**Comment by czy88840616** [comment]
直接使用 otel 组件，打印 ctx.traceId即可。

**Reply by Yuliang-Lee** [reply]
那我是要在整个项目所有 logger.info 里加上 ctx.traceId 吗

**Reply by czy88840616** [reply]
配置context format就好
